Packs/McAfee_ESM_v2/Integrations/McAfee_ESM_v2/McAfee_ESM_v2.py

```diff
@@ -0,0 +1,811 @@
+    elif start_time or end_time:
+        raise ValueError('Invalid set times.')
+    if since != 'CUSTOM' and ' ' in since:
+        start_time, _ = parse_date_range(since, '%Y/%m/%d %H:%M:%S')
+    else:
+        if start_time:
+            start_time = convert_time_format(start_time, difference=difference, to_demisto=False)
+        if end_time:
+            end_time = convert_time_format(end_time, difference=difference, to_demisto=False)
+
+        if start_time and end_time and start_time > end_time:
+            raise ValueError('Invalid set times.')
+    return since, start_time, end_time
+
+
+def list_times_set(list_to_set: List, indexes: List, difference: int = 0) -> List:
+    for i in indexes:
+        if list_to_set[i]:
+            list_to_set[i] = time_format(list_to_set[i], difference=difference)
+    return list_to_set
+
+
+def dict_times_set(dict_to_set: Dict, difference: int = 0) -> Dict:
+    for field in dict_to_set.keys():
+        if dict_to_set[field]:
+            if 'time' in field.lower() or 'date' in field.lower():
+                dict_to_set[field] = time_format(dict_to_set[field], difference=difference)
+            elif isinstance(dict_to_set[field], dict):
+                dict_to_set[field] = dict_times_set(dict_to_set[field], difference)
+            elif isinstance(dict_to_set[field], list):
+                for i in range(len(dict_to_set[field])):
+                    if isinstance(dict_to_set[field][i], dict):
+                        dict_to_set[field][i] = dict_times_set(dict_to_set[field][i], difference)
+    return dict_to_set
+
+
+def time_fields(field_list: List[Dict]) -> list:
+    indexes_list = []
+    for i in range(len(field_list)):
+        if 'time' in field_list[i]['name'].lower() or 'date' in field_list[i]['name'].lower():
+            indexes_list.append(i)
+    return indexes_list
+
+
+def table_times_set(table_to_set: Dict, difference: int = 0) -> Dict:
+    indexes_list = time_fields(table_to_set['columns'])
+    for dict_ in table_to_set['rows']:
+        dict_['values'] = list_times_set(dict_.get('values'), indexes_list, difference)
+    return table_to_set
+
+
+def search_readable_outputs(table: Dict) -> str:
+    if 'columns' in table and 'rows' in table:
+        line_1 = line_2 = '|'
+        for header in table.get('columns', []):
+            line_1 += str(header.get('name')) + '|'
+            line_2 += '--|'
+        rows = table['rows']
+        data: List = [str] * len(rows)
+        for i in range(len(rows)):
+            middle = '~'.join(rows[i].get('values', []))
+            middle = middle.replace('|', '\\|')
+            middle = middle.replace('~', '|')
+            data[i] = f'| {middle} |'
+
+        start = f'Search results\n{line_1}\n{line_2}\n'
+        return start + '\n'.join(data)
+    else:
+        return ''
+
+
+def crate_incident(raw_incidents: List[Dict], alarms: bool) -> List[Dict[str, Dict]]:
+    for incident in range(len(raw_incidents)):
```

Comment:
the loop looks like C...
```suggestion
    for incident in raw_incidents:
```